app/code/Magento/Quote/Api/Data/ShippingAssignmentInterface.php

```diff
@@ -34,6 +34,18 @@ public function getItems();
      */
     public function setItems($value);
 
+
+    /**
+     * @param \Magento\Inventory\Model\PackageInterface[]
+     * @return $this
+     */
+    public function setPackages($packages);
+
+    /**
+     * @return \Magento\Inventory\Model\PackageInterface[]
+     */
+    public function getPackages();
```

Comment:
Moreover there is a get/setShipping method here which returns
\Magento\Quote\Api\Data\ShippingInterface
so, you have also duplication of the shipping address here and in Package.